File: apps/web/src/api/health.ts
// Rust 网关 GET /health 的响应类型
export interface EngineStatus {
  connected: boolean;
  engine?: string;
  version?: string;
  error?: string;
}

export interface HealthResponse {
  status: string;
  service: string;
  version: string;
  analysis_engine: EngineStatus;
}

export async function fetchHealth(): Promise<HealthResponse> {
  const resp = await fetch('/health');
  if (!resp.ok) {
    throw new Error(`HTTP ${resp.status}`);
  }
  return resp.json();
}

File: crates/api/src/routes/health.rs
//! 健康检查：上报 HTTP 网关与 Python 分析引擎状态（引擎离线不影响进程存活）。

use axum::extract::State;
use axum::Json;
use serde_json::{json, Value};

use crate::state::AppState;

pub async fn health(State(state): State<AppState>) -> Json<Value> {
    let engine = match state.engine.ping().await {
        Ok(resp) => json!({
            "connected": true,
            "engine": resp.engine,
            "version": resp.version,
        }),
        Err(e) => json!({
            "connected": false,
            "error": e.to_string(),
        }),
    };

    Json(json!({
        "status": "ok",
        "service": "music-api",
        "version": env!("CARGO_PKG_VERSION"),
        "analysis_engine": engine,
    }))
}
